### PR TITLE
Class performance / performance report condition details upgrade

### LINKED ISSUE #1535: [BUG] improve class performance condition details
**Is your feature request related to a problem? Please describe.**
the details are too long

**Describe the solution you'd like**
maybe a list per metric instead of every class in a different line.

Screen Shot 2022-05-30 at 11 50 46

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1535

### CODE REVIEW

**Comment by JKL98ISR** [comment]
Screen Shot 2022-07-05 at 13 23 12

**Review by matanper — APPROVED** [review]

File: deepchecks/tabular/checks/model_evaluation/performance_report.py [hunk 1]
```diff
@@ -163,156 +164,61 @@ def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
...
 
-    def add_condition_train_test_relative_degradation_not_greater_than(self: PR, threshold: float = 0.1) -> PR:
-        """Add condition that will check that test performance is not degraded by more than given percentage in train.
+    def add_condition_train_test_relative_degradation_less_than(self: PR, threshold: float = 0.1) -> PR:
+        """Add condition - test performance is not degraded by more than given percentage in train.
 
         Parameters
         ----------
-        threshold : float , default: 0.1
+        threshold : float
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/model_evaluation/performance_report.py:184]: you left the default in the signature but removed it from documentation...

File: deepchecks/tabular/checks/model_evaluation/performance_report.py [hunk 2]
```diff
@@ -163,156 +164,61 @@ def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
...
-        than 'threshold'.
+        """Add condition - relative ratio difference between highest-class and lowest-class is less than threshold.
 
         Parameters
         ----------
-        threshold : float , default: 0.3
+        threshold : float
             ratio difference threshold
-        score : str , default: None
+        score : str
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/model_evaluation/performance_report.py:203]: again, defaults in signature but not in docstring

File: deepchecks/tabular/checks/model_evaluation/performance_report.py [hunk 3]
```diff
@@ -163,156 +164,61 @@ def reduce_output(self, check_result: CheckResult) -> Dict[str, float]:
...
         -------
-        PR
+        Self
             instance of 'ClassPerformance' or it subtype
 
         Raises
         ------
         DeepchecksValueError
-            if unknown score function name were passed.
+            if unknown score function name were passed;
```
Comment (nirhutnik) [inline deepchecks/tabular/checks/model_evaluation/performance_report.py:214]: why?

File: tests/conftest.py [hunk 4]
```diff
@@ -86,6 +86,8 @@ def validate_suite_result(
     for check_result in result.results:
         if isinstance(check_result, CheckResult) and check_result.have_conditions():
             for cond in check_result.conditions_results:
+                if cond.category == ConditionCategory.ERROR:
+                    print(check_result.check.name, cond.details)
```
Comment (nirhutnik) [inline tests/conftest.py:90]: Was this left here on purpose?

**Review by nirhutnik — APPROVED** [review]

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**